numpy/core/src/private/npy_longdouble.c

```diff
@@ -0,0 +1,99 @@
+#include <Python.h>
+
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_math.h"
+
+/* This is a backport of Py_SETREF */
+#define NPY_SETREF(op, op2)                      \
+    do {                                        \
+        PyObject *_py_tmp = (PyObject *)(op);   \
+        (op) = (op2);                           \
+        Py_DECREF(_py_tmp);                     \
+    } while (0)
+
+
+/* Heavily derived from PyLong_FromDouble
+ * Notably, we can't set the digits directly, so have to shift and or instead.
+ */
+PyObject *
+PyLong_FromLongDouble(npy_longdouble ldval)
```

Comment:
Alternate name proposal `npy_longdouble_to_PyLong`.

Reply:
Either seems fine. Maybe the `npy_` alternative is safer in case Python3 decides to define this too.